packages/search/addon/components/navi-search-bar.js

```diff
@@ -66,21 +66,24 @@ export default class NaviSearchBarComponent extends Component {
 
     // Perform search on enter press or when there's a search query
     if ((event.keyCode === ENTER_KEY && this.searchQuery.length !== 0) || this.searchQuery.length > 0) {
+      dd.actions.open(event);
       this.launchQuery.perform(this.searchQuery, dd, event);
     }
   }
 
   /**
    * @method launchQuery – Launch search task
    * @param {String} query
-   * @param {Object} dd
+   * @returns {Array} results
    */
-  @keepLatestTask
-  *launchQuery(query, dd, event) {
-    dd.actions.open(event);
-    this.searchResults = yield this.searchProviderService.search.perform(query);
-    if (this.searchResults.length === 0 && query !== '') {
-      this.searchResults = [EMPTY_RESULT];
+  @restartableTask
+  *launchQuery(query) {
+    let results;
+    yield setTimeout(() => {}, DEBOUNCE_MS);
```

Comment:
Use [timeout](http://ember-concurrency.com/api/global.html#timeout)

```suggestion
    yield timeout(DEBOUNCE_MS);
```